pallets/collator-selection/src/lib.rs

```diff
@@ -375,7 +408,18 @@ pub mod pallet {
 				index,
 				<frame_system::Pallet<T>>::block_number(),
 			);
-			Some(Self::assemble_collators())
+
+			let candidates = Self::candidates();
+			let candidates_len_before = candidates.len();
+			let active_candidates = Self::kick_stale_candidates(candidates);
+			let result = Self::assemble_collators(active_candidates.clone());
+			let removed = candidates_len_before - active_candidates.len();
```

Comment:
```suggestion
			let active_candidates = Self::kick_stale_candidates(candidates);
			let active_candidates_len = active_candidates.len();
			let result = Self::assemble_collators(active_candidates);
			let removed = candidates_len_before - active_candidates_len;
```

cloning the length is much cheaper than cloning the whole thing.